portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -69,21 +74,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String email = userService.getCurrentUser().getEmail();
     String comment = getParameter(request, ENTITY_COMMENT_HEADER, /* DefaultValue= */ "");
     long timestamp = System.currentTimeMillis();
+    Optional<Double> preSentiment = getSentiment(comment);
+    Double sentiment = preSentiment.isPresent() ? preSentiment.get() : Double.NaN;
 
     // Respond with a refresh and do local and persistent storage updates.
-    storeComments(name, email, comment, timestamp);
+    storeComments(name, email, comment, sentiment, timestamp);
```

Comment:
Can you just pass the optional double in the function? So you don't need to use double.nan

Reply:
Done